File: 238. Product of Array Except Self/Solution.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        
        prefix=1
        suffix=1
        new=[1]*len(nums)
        for i in range(len(nums)):
            new[i]=prefix
            prefix=nums[i]*prefix
            
        for i in range(len(nums)-1,-1,-1):
            new[i]=suffix*new[i]
            suffix=nums[i]*suffix
        return new

         
           



            
       
            

File: 238. Product of Array Except Self/Solution.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n=nums.length;
        int prefix=1;
        int suffix=1;
        int[] result= new int[n];
        for(int i=0;i<n;i++){
            result[i]=prefix;
            prefix=prefix*nums[i];
        }

        for(int i=n-1;i>=0;i--){
            result[i]*=suffix;
            suffix=suffix*nums[i];
        }
        return result;
    }
}